opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java

```diff
@@ -311,13 +311,20 @@ public void set(
             // Set the property.
             ClassUtil.setFieldValue(project, field, value);
 
-            // Refresh repositories for this project as well.
+            // Refresh field values for project's repositories for this project as well.
             List<RepositoryInfo> riList = env.getProjectRepositoriesMap().get(project);
             if (riList != null) {
                 for (RepositoryInfo ri : riList) {
                     Repository repo = getRepository(ri, CommandTimeoutType.RESTFUL);
 
-                    // set the property
+                    // Set the property if there is one.
+                    try {
+                        ClassUtil.getFieldValue(repo, field);
```

Comment:
Shouldn't the `setFieldValue` be called here and the invocation below removed?

Reply:
Or, if we want to propagate errors, then maybe introduce a new method `hasField` or similar?

Reply:
Definitely want to propagate the errors. While adding `hasField()` I realized the controller actually sets the values on wrong repository object set. Setting the values on `Repository` objects constructed from `RepositoryFactory` will not have any effect on the configuration. The test was wrong also.